src/SparsityPatternView.hpp

```diff
@@ -240,46 +240,18 @@ class SparsityPatternView : protected ArrayOfSetsView< COL_TYPE, INDEX_TYPE >
    *       If USE_ARRAY_BOUNDS_CHECK is defined a lack of space will result in an error,
    *       otherwise the values in the subsequent row will be overwritten.
    */
+  template< typename ITER >
   LVARRAY_HOST_DEVICE inline
-  INDEX_TYPE_NC insertNonZeros( INDEX_TYPE const row, COL_TYPE const * cols, INDEX_TYPE const ncols ) const LVARRAY_RESTRICT_THIS
+  INDEX_TYPE_NC insertNonZeros( INDEX_TYPE const row, ITER const first, ITER const last ) const LVARRAY_RESTRICT_THIS
```

Comment:
Is there an advantage to the iterator interface?

Reply:
It lets you insert directly from a `std::set` for example, it's a lot more flexible than the pointer approach. It won't be faster than the pointer version but the hit should be negligible compared to the cost of inserting.